app/team.go

```diff
@@ -1137,13 +1137,32 @@ func (a *App) GetTeamUnread(teamID, userID string) (*model.TeamUnread, *model.Ap
 	}
 
 	var teamUnread = &model.TeamUnread{
-		MsgCount:     0,
-		MentionCount: 0,
-		TeamId:       teamID,
+		MsgCount:         0,
+		MentionCount:     0,
+		MentionCountRoot: 0,
+		TeamId:           teamID,
+	}
+	migrationNeeded := false
```

Comment:
Would it not be cleaner to extract any migration code (which will live forever, and in the future there may be other reasons for migrations to be added) into a separate (private) method?

Reply:
@zefhemel not sure what you mean. Actual migration is encapsulated in populateMentionCountRoot

Reply:
I mean this line up to 1162 where you track whether migration is required and apply the migration. My suggestion was to extract this code into into a `migrateTeamUnread` method or whatever to keep this method somewhat short. Not a big deal, just an idea.